nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -264,18 +265,19 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             }
             bucketLister.setNextMarker();
 
+            totalListCount += listCount;
             commit(context, session, listCount);
             listCount = 0;
         } while (bucketLister.isTruncated());
+
+        // Update stateManger with the most recent timestamp
         currentTimestamp = maxTimestamp;
+        persistState(context);
 
         final long listMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
         getLogger().info("Successfully listed S3 bucket {} in {} millis", new Object[]{bucket, listMillis});
 
-        if (!commit(context, session, listCount)) {
-            if (currentTimestamp > 0) {
-                persistState(context);
-            }
```

Comment:
Note that this `commit` isn't required, since the last part of the main do/while loop already does a `commit`. Further, it sets `listCount` to zero, so this branch would always be taken.